frame/contracts/src/lib.rs

```diff
@@ -372,6 +380,68 @@ pub mod pallet {
 				T::WeightInfo::on_process_deletion_queue_batch()
 			}
 		}
+
+		fn integrity_test() {
+			// Total runtime memory is expected to have 128Mb upper limit
+			const MAX_RUNTIME_MEM: u32 = 1024 * 1024 * 128;
+			// Memory limits for a single contract:
+			// Value stack size: 1Mb per contract, default defined in wasmi
+			const MAX_STACK_SIZE: u32 = 1024 * 1024;
+			// Heap limit is normally 16 mempages of 64kb each = 1Mb per contract
+			let max_heap_size = T::Schedule::get().limits.max_memory_size();
+			let max_call_depth = T::CallStack::size() as u32;
```

Comment:
hmm, but actually call depth is `CallStack::size()+1`, apparently this check needs to be fixed accordingly

Reply:
Yes. Good catch.